File: ios-swiftui/library/Sources/PlanetComponents/TspAvatar.swift
import SwiftUI

/// Circular avatar — initials or remote image URL.
public struct TspAvatar: View {
    public var text: String
    public var src: String?
    public var size: String
    public var variant: String
    public var theme: StarPlanetTheme

    public init(
        text: String = "",
        src: String? = nil,
        size: String = "md",
        variant: String = "default",
        theme: StarPlanetTheme = .sky
    ) {
        self.text = text
        self.src = src
        self.size = size
        self.variant = variant
        self.theme = theme
    }

    private var dim: CGFloat {
        switch size {
        case "sm": return 32
        case "lg": return 56
        default: return 40
        }
    }

    private var initials: String {
        let trimmed = text.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !trimmed.isEmpty else { return "?" }
        let parts = trimmed.split(whereSeparator: { $0.isWhitespace }).map(String.init)
        if parts.count >= 2 {
            let a = parts[0].prefix(1)
            let b = parts[1].prefix(1)
            return String(a + b).uppercased()
        }
        return String(trimmed.prefix(2)).uppercased()
    }

    private var bg: Color {
        switch variant {
        case "primary": return theme.brandPrimary
        case "subtle": return theme.brandSubtle
        default: return theme.surfaceSubtle
        }
    }

    private var fg: Color {
        switch variant {
        case "primary": return .white
        case "subtle": return theme.brandPrimary
        default: return theme.textPrimary
        }
    }

    public var body: some View {
        Group {
            if let src, !src.isEmpty, let url = URL(string: src) {
                AsyncImage(url: url) { phase in
                    switch phase {
                    case .success(let image):
                        image.resizable().scaledToFill()
                    default:
                        initialsView
                    }
                }
            } else {
                initialsView
            }
        }
        .frame(width: dim, height: dim)
        .background(bg)
        .clipShape(Circle())
        .overlay(Circle().stroke(theme.borderDefault, lineWidth: 1.5))
        .accessibilityLabel(text.isEmpty ? "Avatar" : text)
    }

    private var initialsView: some View {
        Text(initials)
            .font(.system(size: dim * 0.38, weight: .heavy))
            .foregroundColor(fg)
            .frame(maxWidth: .infinity, maxHeight: .infinity)
    }
}

File: react-native/library/src/starPlanet/components/TspAvatar.js
import React from 'react';
import { Image, Text, View, StyleSheet } from 'react-native';
import { withTheme } from '../utils/shared';

const SIZE_MAP = { sm: 32, md: 40, lg: 56 };

/** Circular avatar — initials or remote/local Image `src`. */
export function TspAvatar({
  text = '',
  src,
  size = 'md',
  variant = 'default',
  theme,
}) {
  const t = withTheme(theme);
  const dim = SIZE_MAP[size] || SIZE_MAP.md;
  const initial = String(text || '?').trim().slice(0, 2).toUpperCase();
  const bg =
    variant === 'primary' ? t.brandPrimary : variant === 'subtle' ? t.brandSubtle : t.surfaceSubtle;
  const fg = variant === 'primary' ? '#FFFFFF' : t.textPrimary;

  return (
    <View
      accessibilityRole="image"
      accessibilityLabel={text || 'Avatar'}
      style={[
        local.root,
        {
          width: dim,
          height: dim,
          borderRadius: dim / 2,
          backgroundColor: bg,
          borderColor: t.borderDefault,
        },
      ]}
    >
      {src ? (
        <Image
          source={typeof src === 'string' ? { uri: src } : src}
          style={{ width: dim, height: dim, borderRadius: dim / 2 }}
          accessibilityIgnoresInvertColors
        />
      ) : (
        <Text style={[local.text, { color: fg, fontSize: dim * 0.38 }]}>{initial}</Text>
      )}
    </View>
  );
}

const local = StyleSheet.create({
  root: {
    borderWidth: 1,
    alignItems: 'center',
    justifyContent: 'center',
    overflow: 'hidden',
  },
  text: { fontWeight: '800', textAlign: 'center' },
});
